_examples/postcard/storage/mapping_postgres_eventstore.go

```diff
@@ -86,25 +88,17 @@ type Created struct {
 	ID string `json:"id"`
 }
 
-type CreatedMapper struct{}
-
-func (CreatedMapper) SupportedEvent() stream.Event[postcard.Postcard] {
+func (e Created) SupportedEvent() stream.Event[postcard.Postcard] {
 	return postcard.Created{}
 }
 
-func (CreatedMapper) StorageEvent() any {
-	return Created{}
-}
-
-func (CreatedMapper) ToStorage(event stream.Event[postcard.Postcard]) any {
-	e := event.(postcard.Created)
-	return Created{
-		ID: e.ID,
-	}
+func (e Created) FromEvent(event stream.Event[postcard.Postcard]) transport.Event[postcard.Postcard] {
```

Comment:
let me think about it 🤔

Reply:
https://github.com/ThreeDotsLabs/esja/pull/18/commits/bdf338bf64cff5ebfc21f7a88947c2b3ec6c3238 How about this?

Reply:
Unfortunately, it fails. The problem is that I could not instantiate this event well it is registered as a pointer. Pretty much the same issue as with `stream` interface method `NewFromEvents(events *Events[T]) *T` Maybe I could use a similar naming here? 🤔

Reply:
https://github.com/ThreeDotsLabs/esja/pull/18/commits/fa9c738bbb7d687caa0491af6cd110ee723c0eb0

Reply:
Could we register structs but call the pointer-receiver methods on the newly created type? I think that should be possible 🤔

Reply:
https://github.com/ThreeDotsLabs/esja/commit/79a332e630a0c4815fab2b23daf37871e448ee5d

A rough idea, but it seems to work.

The anonymizer gets into an infinite loop, though. I guess it's because of how `pii` is implemented.

Reply:
I tried exactly the same, but for some reason, it did not work. Let me play with that a bit.

Reply:
Finally! https://github.com/ThreeDotsLabs/esja/pull/18/commits/624dd09e30b89be4436c8910951860e7aebbdcf2

Reply:
Thanks for the suggestion!

Reply:
🤩